clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureResourceManagerClient.groovy

```diff
@@ -86,6 +87,19 @@ class AzureResourceManagerClient extends AzureBaseClient {
     }
   }
 
+  ResourceGroup initializeResourceGroupAndVNet(AzureCredentials creds, String resourceGroupName, String region) {
+    ResourceGroup resourceGroup
+    if (!resourceGroupExists(creds, resourceGroupName)) {
```

Comment:
Per the **_current_** SDK implementation, if the HTTP Status code != OK for the rest call, then it will throw a ServiceException. We could trap for this and then call createOrUpdate in those situations were we get the exception. It would reduce the number of calls to Azure to at most 2, instead of at least 2 as it currently is. I'll take a peek at the new SDK and see if it has the same behavior with throwing an exception.